tests/e2escenarios/e2e_test.go

```diff
@@ -270,4 +272,137 @@ var _ = Describe("E2E Test", func() {
 			Eventually(string(commonVar.CliRunner.Run(getSVCArgs...).Out.Contents()), 60, 3).ShouldNot(ContainSubstring(serviceName))
 		})
 	})
+
+	Context("starting with non-empty Directory add Binding", func() {
+		componentName := helper.RandString(6)
+
+		sendDataEntry := func(url string) map[string]interface{} {
+			values := map[string]interface{}{"name": "joe",
+				"location": "tokyo",
+				"age":      23,
+			}
+			json_data, err := json.Marshal(values)
+			Expect(err).To(BeNil())
+			resp, err := http.Post(fmt.Sprintf("http://%s/api/newuser", url), "application/json", bytes.NewBuffer(json_data))
+			Expect(err).To(BeNil())
+			var res map[string]interface{}
+			err = json.NewDecoder(resp.Body).Decode(&res)
+			Expect(err).To(BeNil())
+			return res
+		}
+
+		receiveData := func(url string) (string, error) {
+			resp, err := http.Get(fmt.Sprintf("http://%s", url))
+			if err != nil {
+				return "", err
+			}
+			defer resp.Body.Close()
+			body, err := io.ReadAll(resp.Body)
+			Expect(err).To(BeNil())
+			return string(body), nil
+		}
+
+		var _ = BeforeEach(func() {
+			commonVar.CliRunner.EnsureOperatorIsInstalled("service-binding-operator")
+			commonVar.CliRunner.EnsureOperatorIsInstalled("cloud-native-postgresql")
+			Eventually(func() string {
+				out, _ := commonVar.CliRunner.GetBindableKinds()
+				return out
+			}, 120, 3).Should(ContainSubstring("Cluster"))
+			helper.Chdir(commonVar.Context)
+			helper.CopyExample(filepath.Join("source", "devfiles", "go"), commonVar.Context)
+			addBindableKind := commonVar.CliRunner.Run("apply", "-f", helper.GetExamplePath("source", "devfiles", "go", "cluster.yaml"))
+			Expect(addBindableKind.ExitCode()).To(BeEquivalentTo(0))
+		})
+
+		It("should verify developer workflow of using binding as env in innerloop", func() {
+			bindingName := helper.RandString(6)
+
+			command := []string{"odo", "init"}
+			_, err := helper.RunInteractive(command, nil, func(ctx helper.InteractiveContext) {
+
+				// helper.ExpectString(ctx, "Based on the files in the current directory odo detected")
+				helper.ExpectString(ctx, "Language: Go")
+				helper.ExpectString(ctx, "Project type: Go")
+				helper.ExpectString(ctx, "Is this correct")
+
+				helper.SendLine(ctx, "")
+
+				helper.ExpectString(ctx, "Select container for which you want to change configuration?")
+
+				helper.SendLine(ctx, "")
+
+				helper.ExpectString(ctx, "Enter component name")
+
+				helper.SendLine(ctx, componentName)
+
+				helper.ExpectString(ctx, "Your new component '"+componentName+"' is ready in the current directory")
+
+			})
+			Expect(err).To(BeNil())
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElement("devfile.yaml"))
+
+			// // "execute odo dev and add changes to application"
+			var devSession helper.DevSession
+			var ports map[string]string
+
+			devSession, _, _, ports, err = helper.StartDevMode(nil)
```

Comment:
Signature has changed in #6332 

```suggestion
			devSession, _, _, ports, err = helper.StartDevMode(helper.DevSessionOpts{})
```